### PR TITLE
AzureMonitor: Prometheus exemplars support 

### PR DESCRIPTION
**Pull request by aangelisc**
This PR adds exemplars support for the Azure Monitor data source.

Users of the data source will now be able to configure it as an internal link option for exemplars when configuring a Prometheus data source. This will allow a users Prometheus monitoring solution to emit exemplar data with the `traceID` of a trace stored in Application Insights and be able to correlate the two from within Grafana.

I've also made some minor changes to how the query editor functions:

- When running a `traceql` query (which is the type for exemplars) the resource selection field is not displayed.
- When the `operationID` field is cleared and the query type is `traceql` the query editor will now actively reset the query to a default traces query (with the correct query type) to allow users to query other resources. This will cause the resource field to be displayed again.

I've also done some minor refactoring and added some helper functions to retrieve the default subscription value when making `traceql` queries.

https://github.com/grafana/grafana/assets/15019026/76d6fc73-69f0-4621-9a4f-0eb3cf87a70b

### CODE REVIEW

File: pkg/tsdb/azuremonitor/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
+)
+
+func GetFirstSubscriptionOrDefault(ctx context.Context, dsInfo types.DatasourceInfo, logger log.Logger) (string, error) {
+	if dsInfo.Settings.SubscriptionId != "" {
+		return dsInfo.Settings.SubscriptionId, nil
+	}
+
+	subscriptionsApiVersion := "2020-01-01"
```
Comment (bossinc) [inline pkg/tsdb/azuremonitor/utils/utils.go:18]: nit this should be a `const`

**Review by bossinc — APPROVED** [review]
Nice! just a few nits

File: pkg/tsdb/azuremonitor/utils/utils.go (hunk 1 above)
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/utils/utils.go]: Have converted this to a `const` and reused in one other place

**Review by bohandley — APPROVED** [review]
LGTM from @grafana/observability-metrics, thank you for this! 🚀

**Comment by kayodeprinceMS** [comment]
@aangelisc can you share a link to the demo app with the requisite instrumentation to repro the demo in the video?

**Comment by aangelisc** [comment]
Hi @kayodeprinceMS, please find an example repository [here](https://github.com/aangelisc/exemplars).

**Pull request merged**